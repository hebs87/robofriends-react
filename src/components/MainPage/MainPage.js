import {Component} from "react";
import Header from "../Header/Header";
import CardList from "../CardList/CardList";
import SearchBox from "../SearchBox/SearchBox";
import Scroll from "../Scroll/Scroll";
import ErrorBoundary from "../ErrorBoundary/ErrorBoundary";
import './MainPage.css';

class MainPage extends Component {

  componentDidMount() {
    this.props.onRequestRobots();
  }

  filteredRobots = robots => {
    return robots.filter(robots => (
      robots.name.toLowerCase().includes(this.props.searchField.toLowerCase())
    ));
  };

  render() {
    const {onSearchChange, robots, isPending} = this.props;

    return (
      <div className='tc'>
        {
          isPending &&
          <h1 className='f1'>Loading...</h1>
        }
        {
          !isPending &&
          <>
            <Header/>
            <SearchBox onSearchChange={onSearchChange}/>
            <Scroll>
              <ErrorBoundary>
                <CardList robots={this.filteredRobots(robots)}/>
              </ErrorBoundary>
            </Scroll>
          </>
        }
      </div>
    )
  }
}

export default MainPage;