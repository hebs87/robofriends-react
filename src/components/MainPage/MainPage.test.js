import {shallow} from 'enzyme';
import MainPage from "./MainPage";

describe('MainPage component test suite', () => {
  // Run this snippet before each test - create mockProps and the wrapper
  let wrapper;
  beforeEach(() => {
    // Create mock props to pass into MainPage component, with mock onRequestRobots function
    const mockProps = {
      onRequestRobots: jest.fn(),
      robots: [],
      searchField: '',
      isPending: false
    }
    wrapper = shallow(<MainPage {...mockProps}/>);
  });

  it('expect to render MainPage component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('filters robots correctly', () => {
    // Create new mockProps2 with robots value to test search/filter functionality
    const mockProps2 = {
      onRequestRobots: jest.fn(),
      robots: [{
        id: 1,
        name: 'John',
        email: 'john@gmail.com'
      }],
      searchField: 'john',
      isPending: false
    }
    const wrapper2 = shallow(<MainPage {...mockProps2}/>);
    // instance() gives us access to all methods in the component
    expect(wrapper2.instance().filteredRobots([])).toEqual([]);
    expect(wrapper2.instance().filteredRobots(mockProps2.robots)).toEqual([{
      id: 1,
      name: 'John',
      email: 'john@gmail.com'
    }]);
  });
});
