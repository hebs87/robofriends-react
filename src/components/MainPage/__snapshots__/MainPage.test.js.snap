// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage component test suite expect to render MainPage component 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <Scroll>
    <ErrorBoundary>
      <CardList
        robots={Array []}
      />
    </ErrorBoundary>
  </Scroll>
</div>
`;
